src/main/java/seedu/address/model/pdfmanager/PdfWrapperLayout.java

```diff
@@ -0,0 +1,157 @@
+            pageHeaderLayout.createPageHeader();
+
+            //create driver layout which contains his tasks for the day.
+            PdfDriverLayout driverLayout = new PdfDriverLayout(document, driver, dateOfDelivery);
+            driverLayout.createDriverDetails();
+
+            //create Table header layout
+            PdfTableHeaderLayout tableHeaderLayout = new PdfTableHeaderLayout();
+            Table headerTable = tableHeaderLayout.createTable();
+            document.add(headerTable);
+
+            //loop through the task list and append this driver's tasks into the layout
+            for (Task task : taskList) {
+                addTaskUnderDriver(driver, task);
+            }
+        }
+    }
+
+    /**
+     * Appends delivery tasks under the driver layout if the tasks belong to him.
+     *
+     * @param driver driver on duty.
+     * @param task task to be append if task belongs to driver.
+     */
+    private void addTaskUnderDriver(Driver driver, Task task) {
+        //since the task list is filtered to only have assigned tasks, then driver WILL NOT be null.
+        assert task.getDriver().isPresent();
+        Driver driverAssigned = task.getDriver().get();
+        if (driverAssigned.equals(driver)) {
+            PdfTaskLayout taskLayout = new PdfTaskLayout(task);
+            Table taskTable = taskLayout.createTable();
+
+            //ensure no splitting tables
+            taskTable.setKeepTogether(true);
+
+            document.add(taskTable);
+        }
+    }
+
+    /**
+     * Sorts driver list accordingly to driver's name in ascending order.
+     */
+    private List<Driver> sortDriverByName(List<Driver> drivers) {
+        Comparator<Driver> driverComparator = Comparator.comparing(d -> d.getName().toString());
+        drivers.sort(driverComparator);
+
+        return drivers;
+    }
+
+    /**
+     * Creates a driver list out of task list.
+     * The driver list contains the drivers working on the specific date.
+     *
+     * @param tasks task list for a specific date.
+     * @return driver list that contains only drivers that is working on the specific date.
+     */
+    private List<Driver> extractDriversFromTaskList(List<Task> tasks) {
+        List<Driver> driverList = new ArrayList<>();
+        for (Task task : tasks) {
+            Driver driver = task.getDriver().get();
+            if (!driverList.contains(driver)) {
+                driverList.add(driver);
+            }
+        }
+
+        return driverList;
+    }
+
+    /**
+     * Filter task list to get tasks on a specific date. Only account for ONGOING and COMPLETE tasks.
+     * INCOMPLETE tasks are not needed as they are not assigned to any drivers.
+     *
+     * @param tasks task list.
+     * @param date  date of delivery.
+     * @return filtered task list that contains only tasks on a specific date and are ONGOING or COMPLETE.
+     */
+    private List<Task> filterTasksBasedOnDate(List<Task> tasks, LocalDate date) {
```

Comment:
This code seems very familiar. If it's anywhere else, consider reusing that code.

Reply:
i try and find